rest_api/aggregation/database_hooks.py

```diff
@@ -121,6 +123,35 @@ def _expression(self, species_contam):
         return sorted(k for k, v in species_contam['contaminant_unique_mapped'].items() if v > 500)
 
 
+class RequiredYield(Calculation):
+    @cached_property
+    def quantised_yields(self):
+        return cfg['available_yields']
+
+    def _expression(self, genome_size):
+        return {
+            # keys have to be strings to be valid BSON
+            str(required_coverage) + 'X': self._get_yield(genome_size, required_coverage)
+            for required_coverage in cfg['available_coverages']
+        }
+
+    def _get_yield(self, genome_size, required_coverage):
+        exact_yield = genome_size * required_coverage
+        for y in sorted(self.quantised_yields):
+            if y >= exact_yield:
+                return y
+
+        raise ValueError(
+            'Could not resolve a required yield for %sX coverage and genome size %s' % (required_coverage, genome_size)
```

Comment:
Not a big fan of the aggregation raising exception. These are not translated in anything the request can understand. We can either:
 - raise an werkzeug exception such as abort. This has the additional disadvantage of leaving the request half processed. 
 - fail silently and log the error and have sensible default such as None or 0.